monailabel/tasks/activelearning/random.py

```
@@ -42,7 +42,27 @@ def __call__(self, request, datastore: Datastore):
         current_ts = int(time.time())
         weights = [current_ts - info.get("ts", 0) for info in images_info]
 
-        image = random.choices(images, weights=weights)[0]
-        logger.debug(f"Random: Images: {images}; Weight: {weights}")
-        logger.info(f"Random: Selected Image: {image}; Weight: {weights[0]}")
-        return {"id": image, "weight": weights[0]}
+        if not any(weights):
+            # All weights are zero (every image was seen at the current second);
+            # fall back to a uniform random pick to avoid a ValueError from random.choices.
+            selected_idx = random.randrange(len(images))
```

Ruff (0.15.5)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)